components/membership/membership-rest-impl/src/main/kotlin/net/corda/membership/impl/rest/v1/NetworkRestResourceImpl.kt

```diff
@@ -41,13 +42,31 @@ class NetworkRestResourceImpl @Activate constructor(
         request: HostedIdentitySetupRequest
     ) {
         val operation = "set up locally hosted identities"
+        val (preferredSessionKey, alternativeSessionKeys) = if (request.sessionKeysAndCertificates.isNotEmpty()) {
+            val preferredSessionKeys = request.sessionKeysAndCertificates.filter {
+                it.preferred
+            }
+            if (preferredSessionKeys.isEmpty()) {
+                throw BadRequestException("No preferred session key was selected.")
+            }
+            if (preferredSessionKeys.size > 1) {
+                throw BadRequestException("Can not have more than one preferred session key.")
+            }
+            val preferredSessionKey = preferredSessionKeys.first()
+            val alternativeSessionKeys = request.sessionKeysAndCertificates.filter {
+                it != preferredSessionKey && !it.preferred
+            }
```

Comment:
```suggestion
            val alternativeSessionKeys = request.sessionKeysAndCertificates.filter { !it.preferred }
```
wouldn't this be sufficient?